src/System.Memory/ref/System.Memory.cs

```diff
@@ -117,5 +117,86 @@ public static class SpanExtensions
         
         public static ReadOnlySpan<TTo> NonPortableCast<TFrom, TTo>(this ReadOnlySpan<TFrom> source) where TFrom : struct where TTo : struct { throw null; }
     }
+
+    [System.Runtime.InteropServices.StructLayoutAttribute(System.Runtime.InteropServices.LayoutKind.Sequential)]
+    public struct ReadOnlyMemory<T>
+    {
+        public static ReadOnlyMemory<T> Empty { get { throw null; } }
+        public ReadOnlyMemory(T[] array) { throw null;}
+        public ReadOnlyMemory(T[] array, int start, int length) { throw null;}
+        internal ReadOnlyMemory(Buffers.OwnedMemory<T> owner, int index, int length) { throw null;}
+        public bool IsEmpty { get { throw null; } }
+        public int Length { get { throw null; } }
+        [System.ComponentModel.EditorBrowsable(System.ComponentModel.EditorBrowsableState.Never)]
+        public override bool Equals(object obj) { throw null; }
+        public bool Equals(ReadOnlyMemory<T> other) { throw null; }
+        [System.ComponentModel.EditorBrowsable(System.ComponentModel.EditorBrowsableState.Never)]
+        public override int GetHashCode() { throw null; }
+        public static implicit operator ReadOnlyMemory<T>(T[] array) { throw null; }
+        public static implicit operator ReadOnlyMemory<T>(ArraySegment<T> arraySegment) { throw null; }
+        public ReadOnlyMemory<T> Slice(int start) { throw null; }
+        public ReadOnlyMemory<T> Slice(int start, int length) { throw null; }
+        public ReadOnlySpan<T> Span { get { throw null; } }
```

Comment:
Should this be AsReadOnlySpan?

Reply:
I have to say I am not a big fan of this whole AsReadOnlyXxx naming pattern. I would prefer all APIs to get spans or memory to be simple/same property/method and the return type being the right type (read-only or r/w) depending on the source. Otherwise it's more difficult to move code from r/w to r/o, and the caller has to continuesly remember which variables are r/o vs r/w

Reply:
Also, after we removed all AsSpan methods that take parameters, I am not sure if the value from calling the API AsSpan and making it a method is there. simple Span property is so much more elegant. 

Having said that I feel less strongly about this comment because there is some value in the API being a method. Some conversions to span will not be instance members; they will be extensions methods. And so AsSpan() method can be used for both instance and extension methods.

Too bad we don't have extensions properties yet. cc: @MadsTorgersen, @terrajobst, @jaredpar

Reply:
I wasn't suggesting changing it from a property to a method. We have the AsMemory property on OwnedMemory, so I thought maybe we should call the Span property AsSpan. Is the convention that AsX should be a method?

I had changed Memory to AsMemory based on @stephentoub's feedback here:
https://github.com/dotnet/coreclr/pull/13583#discussion_r135241523

Reply:
I am fine with either way for now, but I think we should sit down and think where this AsXxx convention ends. are we now prefixing all properties or factory methods with "As"? e.g. Should DateTimeOffset.Date be called AsDateTime? What about String.Chars? Should it be called AsChars? Should Uri.OriginalString be called AsOriginalString?

@terrajobst, @stephentoub

Reply:
> I am fine with either way for now

Just to be clear, my feedback comment was:
> Also, from a naming perspective, why is it AsSpan but just Memory rather than having Span/Memory or AsSpan/AsMemory?
In other words, I don't care whether there's an `As` prefix or not, just that we're consistent as to whether there is one.

> What about String.Chars?

What is String.Chars?  Did you mean String.ToCharArray?

> Should Uri.OriginalString be called AsOriginalString?

OriginalString isn't converting Uri to an original string: it's returning the original string that was passed in.

Reply:
Yeah, I agree that some of the examples I gave are different than Memory.AsSpan too illustrate that we need guidance to know where we go to far with using the As prefix.

Reply:
Regardless of what we decide, I don't think properties should use "As" or "To" prefixes; historically I believe we've reserved those for methods.  So these should be "AsSpan()/AsMemory()" or "Span/Memory".